Alternative layout definition with flex

corwin-of-amber opened the pull request:
I played around with the layout a bit this evening.

It looks and behaves almost like the previous layout (that uses "fixed" and padding), but fixes some issues such as the vertical scrollbar appearing to the right of the panel.
Also a max-width of 37em was set for the panel.
(it's a bit arbitrary, but the `#document` is set to 51em, so I said meh)

ejgallego commented:
Very cool! At some point I think Benoît Pin had something like that, it seems we didn't manage to integrate it.

I've added a "needs: testing" flag to signal PRs that are only not merged because it'd be good to get some testing on them first.

corwin-of-amber commented:
Yes, esp. we need to test on different operating systems. I tried on my Mac with Safari, Chrome, and Firefox. You can try Linux, we can both try our Androids and I have VMs running Windows 7 & 10 -- I bet Edge is out of the question but Chrome and FF on Windows should work.

corwin-of-amber commented:
Correction: seems to have broken printing in FF. The second page is empty.

ejgallego commented on ui-css/coq-base.css:
```diff
@@ -48,27 +53,31 @@ a, a:link, a:visited, a:active {
-  position: fixed;
+  position: relative; /* for the power button */
   width: 45%;
+  max-width: 37em;
```
Umm, this makes the panel to display too small in my setup. Why do you want to constraint max-width to such a low number?

ejgallego replied:
Oh I just was the comment, IMHO we want to tweak it out a bit, it doesn't look good in my laptop [1444p]

ejgallego commented:
Thanks @corwin-of-amber , it looks pretty good, but indeed I have a couple of display issues; I'd like to try to refine it a bit if you don't mind, I guess merging this is not urgent, right?

I'll be very busy for the rest of the next week [until July] so please bear with me.

corwin-of-amber commented:
Definitely not urgent, and `max-width: 37em` is not the crucial piece anyway. Just thought it would be a nice touch but obviously I was a bit careless with the number.

On a high-resolution display (e.g. connected to my monitor, 3840px across) the panel seemed to take up a rather ridiculous amount of space, most of which goes unused. Can try to fine-tune with `@media` selectors so that it only applies to such scenarios.

corwin-of-amber commented:
Perhaps `em` is not a good unit of measurement? I thought it would be ok since it is proportional to the font size. On my MacBook Pro (retina), there is a very small difference between having `max-width: 37em` and not having it.

<img width="1507" alt="Screen Shot 2019-06-22 at 22 01 05" src="https://user-images.githubusercontent.com/704372/59967871-5304aa80-9539-11e9-84c0-0639489c1f96.png">

<img width="1507" alt="Screen Shot 2019-06-22 at 22 01 26" src="https://user-images.githubusercontent.com/704372/59967881-8f380b00-9539-11e9-9176-e2e0ca50f22e.png">

ejgallego commented:
After playing with this again, I see advantages to both setups, so if it is possible I'd prefer we add "theming" support and indeed do allow both layouts .

corwin-of-amber commented:
Ok, I will add this option in JS and update the README and CHANGES.

I suppose the existing layout should remain the default then?

ejgallego commented:
> I suppose the existing layout should remain the default then?

No strong opinion on that one, indeed you actually have more experience as to evaluate how they work for users. Users of the old layout can just set the variable in `config` so IMHO not a big deal.

corwin-of-amber commented:
Added the config option + docs and rebased.

I noticed that `options` has a key for the IDE wrapper element id, however the CSS has `#ide-wrapper` hard-coded. We should probably fix that at some point. Setting `wrapper_id` to anything other than the default will not produce a working layout.

ejgallego commented:
> I noticed that `options` has a key for the IDE wrapper element id, however the CSS has `#ide-wrapper` hard-coded. We should probably fix that at some point. Setting `wrapper_id` to anything other than the default will not produce a working layout.

Yup , there was some start to make the code into proper UI components but it got postponed by other issues.
